### PR TITLE
[tests-only] ApiTests. add more search tests

### PR DESCRIPTION
**Pull request by ScharfViktor**
I added some cases: 

- find hidden/deleted/pending/declined folders
- find folder- shares mountpoint
- find project space by name (#4506)

### CODE REVIEW

File: tests/acceptance/features/apiSpaces/search.feature [hunk 1]
```diff
@@ -50,3 +50,51 @@ Feature: Search
       | /SubFolder1                                |
       | /SubFolder1/subFOLDER2                     |
       | /SubFolder1/subFOLDER2/insideTheFolder.txt |
+    And for user "Brian" the search result should contain space "mountpoint/folder"
```
Comment (SagarGi) [inline tests/acceptance/features/apiSpaces/search.feature:53]: Brian seems to be searching for `folder` but this step says search result should contain `space`?
or may be `And for user "Brian" the search result should contain folder "folder" inside of space "Shares Jail"` 
 this type of step would fit in here? @ScharfViktor what do you think?

File: tests/acceptance/features/apiSpaces/search.feature [hunk 2]
```diff
@@ -50,3 +50,51 @@ Feature: Search
...
+    When user "Alice" searches for "folder" using the WebDAV API
+    Then the HTTP status code should be "207"
+    And the search result should contain "0" entries
+    
+
+  Scenario: User can find project space by name
+    When user "Alice" searches for "find data" using the WebDAV API
+    Then the HTTP status code should be "207"
+    And the search result should contain "1" entries
+    And for user "Alice" the search result should contain space "find data"
```
Comment (SagarGi) [inline tests/acceptance/features/apiSpaces/search.feature:99]: In here `alice` is searching for `space` which seems good.

File: tests/acceptance/features/apiSpaces/search.feature (hunk 1 above)
Comment (ScharfViktor) [reply tests/acceptance/features/apiSpaces/search.feature:53]: Sorry that I didn't give you more context https://github.com/owncloud/ocis/pull/4463#issue-1351837237

**Review by phil-davis — APPROVED** [review]

File: tests/acceptance/features/apiSpaces/search.feature (hunk 1 above)
Comment (ScharfViktor) [reply tests/acceptance/features/apiSpaces/search.feature:53]: `And for user "Brian" the search result should contain folder "folder" inside of space "Shares Jail"` 
From the web's point of view, it looks the right thing to do, but `folder` is not realy folder, it's space (type mountpoint). So I search result gives me space. I would suggest to keep my version

**Comment by phil-davis** [comment]
Note: I just rebased PR #4546 because of conflicts in SpacesContext. Probably it will be good to wait until that is merged, then rebase this PR to resolve conflicts - we need to edit `SpacesContext` one-at-a-time.

**Comment by ScharfViktor** [comment]
> Note: I just rebased PR #4546 because of conflicts in SpacesContext. Probably it will be good to wait until that is merged, then rebase this PR to resolve conflicts - we need to edit `SpacesContext` one-at-a-time.

Okay, I did a php style commit and interrupted the CI. Then I'm waiting for https://github.com/owncloud/ocis/pull/4546

**Pull request merged**